File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.c
int max(int a, int b) {
    return a > b ? a : b;
}

int longestSubarray(int* nums, int numsSize) {
    int max_val = 0, ans = 0, curr = 0;
    while (numsSize--) {
        if (max_val < *nums) {
            max_val = *nums;
            curr = ans = 0;
        }

        curr = *nums++ == max_val ? curr + 1 : 0;

        ans = max(ans, curr);
    }
    return ans;
}

File: 2503-longest-subarray-with-maximum-bitwise-and/2503-longest-subarray-with-maximum-bitwise-and.kt
class Solution {
    fun longestSubarray(nums: IntArray): Int {
        var max_val: Int = 0; var ans: Int = 0; var curr: Int = 0;
        for (n in nums) {
            if (max_val < n) {
                max_val = n;
                curr = 0; ans = 0;
            }

            curr = if (n == max_val) curr + 1 else 0;

            ans = max(ans, curr);
        }
        return ans;
    }
}